### PR TITLE
Transition Anomaly Detector to Test Proxy

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Transition `azure-ai-anomalydetector` to Test Proxy instead of using local playback records.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/anomalydetector/azure-ai-anomalydetector/pom.xml [hunk 1]
```diff
@@ -38,8 +38,8 @@
   </developers>
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <jacoco.min.linecoverage>0.2</jacoco.min.linecoverage>
```
Comment (mssfang) [inline sdk/anomalydetector/azure-ai-anomalydetector/pom.xml:41]: do you know why to downgrade the version?

**Review by samvaity — APPROVED** [review]
Tets proxy changes lgtm!

File: sdk/anomalydetector/azure-ai-anomalydetector/src/test/resources/session-records/testDetect.json [hunk 2]
```diff
@@ -1,25 +0,0 @@
-{
```
Comment (samvaity) [inline sdk/anomalydetector/azure-ai-anomalydetector/src/test/resources/session-records/testDetect.json:1]: This file did not make it to the assets repo, intentional?

Comment (alzimmermsft) [reply sdk/anomalydetector/azure-ai-anomalydetector/src/test/resources/session-records/testDetect.json:1]: This was an old version of recording that was no longer used but never deleted.

File: sdk/anomalydetector/azure-ai-anomalydetector/pom.xml (hunk 1 above)
Comment (alzimmermsft) [reply sdk/anomalydetector/azure-ai-anomalydetector/pom.xml:41]: I'm not sure why the code coverage became less

**Pull request merged**